packages/patternfly-4/react-core/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import styles from '@patternfly/patternfly/components/Check/check.css';
+import { css, getModifier } from '@patternfly/react-styles';
+import { Omit } from '../../helpers/typeUtils';
+
+export interface CheckboxProps
+  extends Omit<React.HTMLProps<HTMLInputElement>, 'type' | 'onChange' | 'disabled' | 'label'> {
+  className?: string;
+  isValid?: boolean;
+  isDisabled?: boolean;
+  isChecked?: boolean;
+  checked?: boolean;
+  onChange?(checked: boolean, event: React.FormEvent<HTMLInputElement>): void;
+  label?: React.ReactNode;
+  id: string;
+  'aria-label': string;
+}
+
+export const Checkbox: React.FunctionComponent<CheckboxProps> = ({
+  className = '',
+  isValid = true,
+  isDisabled = false,
+  isChecked = null,
+  checked = false,
+  onChange = () => undefined,
+  label = undefined,
+  'aria-label': ariaLabel = '',
+  ...props
+}) => (
+  <div className={css(styles.check, className)}>
+    <input
+      {...props}
+      className={css(styles.checkInput)}
+      type="checkbox"
+      onChange={(event: React.FormEvent<HTMLInputElement>) => {
```

Comment:
Try `onChange={(event: React.FormEvent<HTMLInputElement>) => this.props.onChange(event.currentTarget.checked, event)}`